src/main/java/net/snowflake/client/core/SFStatement.java

```diff
@@ -119,6 +119,10 @@ private SFBaseResultSet executeQuery(
 
     // snowflake specific client side commands
     if (isFileTransfer(trimmedSql)) {
+      if (this.session.isFileTransferDisabled()) {
+        throw new SnowflakeSQLException(
+            ErrorCode.INVALID_SQL, "file transfers are disabled for this connection");
```

Comment:
I wonder if ` UNSUPPORTED_STATEMENT_TYPE_IN_EXECUTION_API(200042, SqlState.FEATURE_NOT_SUPPORTED)` is more appropriate, but probably doesn't matter much for us, more for the snowflake folks.

Reply:
Yeah, I was going back and forth on that, but ultimately I think they'll want to allocate a new error code, so I'll leave that for when I open a PR on upstream